### PR TITLE
Feature: Timeline UI

### PR DESCRIPTION
**Pull request by NghiaTranUIT**
### ⚠️ Warnings
Please don't merge this ticket until all sub-tickets are done. 

- [x] View Selections Tabs: #3015
- [x] Header: #3016
- [x] Improvement Timeline Ticket
- [x] Activity Timeline
- [x] Timeline Selection Border
- [x] Timeline More Info Bubble

### 📒 Description
This ticket is responsible as a parent ticket for the entire **Timeline** feature. All children tickets will branch off and merge from this branch.

### 🕶️ Types of changes
**New feature** (non-breaking change which adds functionality)
**Breaking change** (fix or feature that would cause existing functionality to change)

### 👫 Relationships
https://github.com/toggl-open-source/toggldesktop/projects/4

### 🔎 Review hints
- Review all children tickets separately. Make sure all tickets merge and work well.

### CODE REVIEW

File: src/context.cc [hunk 1]
```diff
@@ -644,7 +648,7 @@ void Context::updateUI(const UIElements &what) {
 
         if (what.display_time_entries && user_) {
             if (what.open_time_entry_list) {
-                time_entry_editor_guid_ = "";
+                UI()->resetTimeEntryGUID();
```
Comment (NghiaTranUIT) [inline src/context.cc]: @IndrekV It's my hot-fix https://github.com/toggl/toggldesktop/commit/9ca209f4b493fee8ea76e3b68200591a2138bd8c for this issue (https://github.com/toggl/toggldesktop/pull/3033#issuecomment-502993150)

The original code forgot to reset the TimeEntryEditorGUID, so it cases the Editor doesn't appear if we select same Time Entry

**Comment by IndrekV** [comment]
`timeline-ui` branch needs some rebasing as it still does not have some fixes. While trying to create a build I get the error we fixed some time ago `Setter for 'textColor' with Objective-C selector 'setTextColor:' conflicts with met`

**Comment by NghiaTranUIT** [comment]
@IndrekV please skip this `textColor` issue temporary. This issue was introduced when I tried to rebased incorrectly the this branch in the past, but it's up-to-date from `master` (according from `master history`)

**Comment by IndrekV** [comment]
I'm not able to create a build from this branch as it always fails with the settext error. It fails when I create it with nightly script and also when I try to create an archive in xcode. Are you able to create a build from this branch?

**Comment by IndrekV** [comment]
First testing build with timeline feature
[TogglDesktop-TimelineUI.zip](https://github.com/toggl/toggldesktop/files/3595726/TogglDesktop-TimelineUI.zip)

**Comment by NghiaTranUIT** [comment]
## Experiment (Shouldn't use in Production)
- UI Improvement (https://github.com/toggl/toggldesktop/issues/3380)
- Dismiss the Timeline Hover when the mouse goes out the pillar (Discussion at https://toggl.slack.com/archives/CH7DV7VSR/p1570380680003800?thread_ts=1570128851.001700&cid=CH7DV7VSR)

Download: [TogglDesktop.zip](https://github.com/toggl/toggldesktop/files/3695658/TogglDesktop.zip)

**Comment by NghiaTranUIT** [comment]
## Timeline Update 1
- The stable Timeline features
- UI Improvement (https://github.com/toggl/toggldesktop/issues/3380)

[TogglDesktop.zip](https://github.com/toggl/toggldesktop/files/3695921/TogglDesktop.zip)

**Comment by NghiaTranUIT** [comment]
This commit https://github.com/toggl/toggldesktop/commit/8648b24f71bec71d0772249175269e91b6928298 introduce a bug when I couldn't close the TE Editor when selecting the TE in the list, which is different behavior than the `master`. I would revert it before doing a release 👍 

Technically, 
```c++
if (UI()->TimeEntryEditorGUID() == te->GUID()) { }
```
never hit true since the `UI()->TimeEntryEditorGUID` always equals to "".

**Comment by NghiaTranUIT** [comment]
## Timeline Update 2
- Version 7.4.1036
- Updated Latest `master` (https://github.com/toggl/toggldesktop/commit/eb7230765c7ba77940cd35f7062d93548422c20d)
- Touch Bar features
- Pre-Populate Timeline Data in Help Menu -> Import Data 

Download: [TogglDesktop.zip](https://github.com/toggl/toggldesktop/files/3808758/TogglDesktop.zip)

**Comment by NghiaTranUIT** [comment]
## Timeline Update 3
- Updated new Timeline Event

Download: 
[TogglDesktop.zip](https://github.com/toggl/toggldesktop/files/3827333/TogglDesktop.zip)

**Comment by NghiaTranUIT** [comment]
## Timeline Update 4
- Zoom with keyboard
- Update from 7.4.1092 beta version.

Download:
[TogglDesktop_Timeline_7.4.1092.zip](https://github.com/toggl-open-source/toggldesktop/files/3939094/TogglDesktop_Timeline_7.4.1092.zip)

**Pull request merged**